### PR TITLE
Add Footer and set up Redux

### PR DESCRIPTION
**Pull request by chrismekim**
### Description

-**NOTE: Actual fonts are not in, and newsletter signup functionality will be done in #20**
- Adds Redux (implemented like Cocoon)
- Adds Footer (Note: Resources and Connect are not buttons, but Companies, About Us, and Why We Invest are buttons.)
- Updates Global Settings schema to include Footer Menu links.

### Type(s) of changes

- [x] New feature

### Motivation for PR

#19 
### Applicable screenshots:

image
image

### Follow-up PR

- Footer Newsletter signup functionality in #49 
- #49 also fixes a Safari height issue with the Footer

### CODE REVIEW

File: src/components/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
+import React from 'react';
+import { getYear } from 'date-fns';
```
Comment (mokaymokay) [inline src/components/Footer.tsx:2]: let's do `import getYear from 'date-fns/get_year'` so we don't import the entire library if we don't need to

File: src/components/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            <div className="col-6 flex flex-col">
+              <Button
+                className="color-charcoal transition-shorter hover-lighten-charcoal secondary-bold-sm items-center bg-color-transparent text-decoration-none mb1_5 lg:mb3_75"
+                ariaLabel={Language.t('Footer.aboutUsButtonAriaLabel')}
+                to={RouteMap.ABOUT.path}
+                label={Language.t('Footer.aboutUs')}
+              />
+              <div className="Footer__sublinks flex flex-col">
+                {footerMenu.aboutLinks.map((link: IButton) => (
+                  <Button
```
Comment (mokaymokay) [inline src/components/Footer.tsx:58]: needs a key

File: src/components/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+            <div className="col-6 flex flex-col">
+              <Button
+                className="color-charcoal transition-shorter hover-lighten-charcoal secondary-bold-sm items-center bg-color-transparent text-decoration-none mb1_5 lg:mb3_75"
+                ariaLabel={Language.t('Footer.whyWeInvestButtonAriaLabel')}
+                to={RouteMap.WHY_WE_INVEST.path}
+                label={Language.t('Footer.whyWeInvest')}
+              />
+              <div className="Footer__sublinks flex flex-col">
+                {footerMenu.whyWeInvestLinks.map((link: IButton) => (
```
Comment (mokaymokay) [inline src/components/Footer.tsx:78]: here as well

File: src/components/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+      <div className="Footer__section-two bg-color-lilac-lighter lg:col-3 px1_5 py3_75 flex flex-row lg:flex-col relative">
+        <div className="flex flex-col col-6 lg:col-12">
+          <div className="color-charcoal secondary-bold-sm mb1_5 lg:mb3_75">
+            {Language.t('Footer.resources')}
+          </div>
+
+          <div className="Footer__sublinks flex flex-col">
+            {footerMenu.resourcesLinks.map((link: IButton) => (
+              <Button
```
Comment (mokaymokay) [inline src/components/Footer.tsx:106]: key

File: src/constants/Language/Locales/en-US.json [hunk 5]
```diff
@@ -1,7 +1,23 @@
...
+    "aboutUs": "About Us",
+    "aboutUsButtonAriaLabel": "Navigate to About Us page",
+    "whyWeInvest": "Why We Invest",
+    "whyWeInvestButtonAriaLabel": "Navigate to Why We Invest page",
+    "resources": "Resources",
+    "resourcesButtonAriaLabel": "Navigate to Resources page",
+    "connect": "Connect",
+    "copyrightLabel": "© %{year}",
+    "allRightsReserved": "All Rights Reserved",
+    "bbgv": "BBG Ventures, LLC"
```
Comment (mokaymokay) [inline src/constants/Language/Locales/en-US.json:20]: Some of these might be repeated in the Nav.. maybe can move to global?

**Review by mokaymokay — APPROVED** [review]
reduxxx

File: src/constants/Language/Locales/en-US.json (hunk 5 above)
Comment (chrismekim) [reply src/constants/Language/Locales/en-US.json:20]: Ahhh very true, ty!

File: src/components/Footer.tsx (hunk 1 above)
Comment (chrismekim) [reply src/components/Footer.tsx:2]: lolll redux does add SO MUCH MORE CODE! :-O 

and good point about importing the entire library here!

File: src/components/Footer.tsx (hunk 2 above)
Comment (chrismekim) [reply src/components/Footer.tsx:58]: ty for the key reminders!!!

File: src/lib/action/promiseActionHandler.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    console.warn(`Promise Rejected: ${action.type}`, action.payload);
+  }
+
+  // if (process.env.NODE_ENV === 'production') {
+  //   Sentry.withScope((scope: Sentry.Scope) => {
+  //     scope.setExtra('reduxAction', action);
+  //     scope.setExtra('reduxState', state);
+  //     Sentry.captureException(action.payload);
+  //   });
+  // }
```
Comment (limsohee1002) [inline src/lib/action/promiseActionHandler.ts:15]: Remove this or add comments

**Review by limsohee1002 — APPROVED** [review]

File: src/lib/action/promiseActionHandler.ts (hunk 6 above)
Comment (chrismekim) [reply src/lib/action/promiseActionHandler.ts:15]: ahh thank you!

**Pull request merged**